src/build-data/buildh.in

```diff
@@ -134,16 +176,17 @@
 * is why this option is not enabled by default.
 */
 #if defined(BOTAN_HAS_VALGRIND) || defined(BOTAN_ENABLE_DEBUG_ASSERTS)
+   /** Memory_Pool will use OS calls to mlock its memory */
```

Comment:
This comment is incorrect. `Memory_Pool` will always try to use `mlock` or some equivalent. The MMU protections are that we will for pages which are currently unused (in the free list) set them to noaccess using `mprotect` or `VirtualProtect`. Thus converting a use-after-free into a crash bug.

Unfortunately doing this is rather expensive which is why it is disabled most of the time :/

Reply:
👍
 
 I simply moved the existing docsting down to this define and added a `@brief` section. Please make sure its correct now and merge at will.